week-9/Homework/mandatory/2-exercises/4-music-player.js

```diff
@@ -20,34 +20,50 @@ This means the order the songs are played in will be random, but each song will
 
 
 class MusicPlayer {
-    // Add your code here
+    constructor() {
+        this.playing = [];
+        this.MusicPlayer = [];
+        this.index = 0;
+    }
+    add(playing, player) {
+        this.playing.push(playing);
+        this.MusicPlayer.push(player);
+
+    }
+    play() {
+        console.log("Currently playing: " + this.playing[this.index] + " by " + this.MusicPlayer[this.index]);
+
+    }
+    skip() {
+        if (this.index < this.playing.length)
+            this.index++;
+        else this.index = 0;
+        console.log("Currently playing: " + this.playing[this.index] + " by " + this.MusicPlayer[this.index]);
+    }
+    previous() {
+        if (this.index < this.playing.length)
+            this.index--;
+        else this.index = this.playing.length - 1;
+
+        console.log("Currently playing: " + this.playing[this.index] + " by " + this.MusicPlayer[this.index]);
```

Comment:
Like above, after setting the index, you simply need to call the play function that will do the rest for you as it is responsible to play the song. Remember the DRY rule!